### PR TITLE
[WIP] Fix #7: Write a simple test suite

### LINKED ISSUE #7: There's no tests :disappointed:

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

**Comment by michaelpj** [comment]
Personally, I'd go for a more minimalist approach: write it in Haskell, make some tiny test repository sources as examples, try to just make some assertions in Haskell about the output index tarball or whatever.

File: tests/FoliageSpec.hs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module FoliageSpec where
+
```
Comment (yvan-sraka) [inline tests/FoliageSpec.hs:2]: I feel dumb here: I struggle a bit to import `Foliage` so I can write tests :/

Comment (michaelpj) [reply tests/FoliageSpec.hs:2]: The problem is that everything was previously defined in an executable. You want to put it in a library so you can import it.

File: foliage.cabal [hunk 2]
```diff
@@ -38,32 +65,23 @@ executable foliage
...
         Network.URI.Orphans
 
-    default-language:   GHC2021
+executable foliage
+    import:             shared-modules
+    main-is:            Main.hs
 
-    ghc-options:        -Wall
-    ghc-options:        -threaded
+library libfoliage
```
Comment (michaelpj) [inline foliage.cabal:72]:
```suggestion
library
```
I think you can do this, executables and libraries don't occupy the same namespace.

File: foliage.cabal [hunk 3]
```diff
@@ -38,32 +65,23 @@ executable foliage
...
+    import:             shared-modules
```
Comment (michaelpj) [inline foliage.cabal:73]: I don't think you want to do this. Rather, inline `shared-modules` into here, put the `Foliage` modules in `exposed-modules`, and add `foliage` to the `build-depends` of the executable.

**Review by michaelpj — COMMENTED** [review]
Have a look at another Haskell package with libraries and executables for inspiration if you get stuck!

File: foliage.cabal (hunk 3 above)
Comment (andreabedini) [reply foliage.cabal]: The idea of writing a library scares me a bit :P I'd be ok with just re-listing the `other-modules`. I am not sure if this is what @yvan-sraka was doing.

Comment (michaelpj) [reply foliage.cabal]: What's wrong with a library? It's just a way of organizing the code really!

Comment (andreabedini) [reply foliage.cabal]: Then I need to care about API changes :D There's no API if nobody else can use the code :D

**Pull request closed without merging**